tiledb/common/logger.h

```diff
@@ -29,36 +29,25 @@
  *
  * This file defines class Logger, which is implemented as a wrapper around
  * `spdlog`. By policy `spdlog` must remain encapsulated as an implementation
- * and not be exposed as a dependency of the TileDB library. Accordingly, this
- * header should not be included as a header in any other header file. For
- * inclusion in a header (notably for use within the definition of
- * template-dependent functions), include the header `logger_public.h`.
- *
- * The reason for this restriction is a technical limitation in template
- * instantiation. Part of the interface to `spdlog` consists of template
- * functions with variadic template arguments. Instantiation of such function
- * does not instantiate a variadic function (for exmaple `printf`) but rather a
- * function with a fixed number of arguments that depend upon the argument list.
- * Such variadic template argument lists cannot be forwarded across the
- * boundaries of compilation units, so exposing variadic template arguments
- * necessarily exposes the dependency upon `spdlog`. Thus this file `logger.h`,
- * which does have such arguments, must remain entirely within the library, but
- * `logger_public.h`, which does not have such arguments, may be exposed without
- * creating an additional external dependency.
+ * and not be exposed as a dependency of the TileDB library.
  */
 
 #pragma once
 #ifndef TILEDB_LOGGER_H
 #define TILEDB_LOGGER_H
 
-#include <spdlog/spdlog.h>
+#include <fmt/format.h>
```

Comment:
Can I `#include <format>` here instead?

Reply:
That would be preferable, but it might not compile on Mac, which at least in the past did not ship with `<format>`. Try it and see.

Reply:
I tried but CI failed everywhere except VS2022. Reverted.